recipe/2023-11-support-insecure-registries.md

```diff
@@ -0,0 +1,185 @@
+# Adding Support for publishing recipes to insecure registries
+
+* **Status**: Pending
+* **Author**: Vishwanath Hiremath (@vishwahiremat)
+
+## Overview
+
+Today we publish/pull recipes to/from the registry that is OCI compliant and only support secure connections i.e registries that are SSL/TLS enabled. And we don't have support to use insecure registry e.g locally hosted k3d registry. And we hit _"http: server gave HTTP response to HTTPS client"_ error if insecure registry is used. This document is centered around addressing this issue.
+
+## Terms and definitions
+
+| Term        | Definition |
+| :-- |:--|
+| insecure registry      | An insecure registry is the one not having either valid registry certificate or is not using TLS and does not use secure communication protocols like HTTPS. |
+| ORAS      | ORAS is the de facto tool for working with OCI Artifacts.https://oras.land/docs/ |
+| Recipes      | Recipes enable a separation of concerns between infrastructure operators and developers by automating infrastructure deployment. https://docs.radapp.dev/author-apps/recipes/ |
+
+## Objectives
+
+> **Issue Reference:** https://github.com/radius-project/radius/issues/6648
+
+### Goals
+
+-   Enable the support to publish and pull recipes using a non-secure registry.
+
+### Non goals
+
+
+### User scenarios (optional)
+
+#### User story 1
+As a Radius user I would like to use locally hosted insecure registry to publish bicep recipes.
+
+Scenario: I create a k3d managed cluster running:
+```
+k3d registry create myregistry.localhost --port 5000
+```
+And use localhost:5000 as the registry path to publish bicep recipes.
+```
+rad bicep publish --file ./redis-test.bicep --target br:localhost:5000/myregistry/redis-test:v1
+```
+I would like to see `rad bicep publish` command successfully publish bicep file to locally hosted registry and also I should be able to read it from the registry during recipe deployment. 
+
+
+## Design
+
+### Design details
+Since we use ORAS client to communicate with the registry, it provides an option "plain-http" to allow insecure connections to registry without SSL check. `plainHttp` property should be set to `true` when client to the remote repository is created to support using insecure registries.
+
+#### Introduce "insecure-http" opt-in flag for users.
```

Comment:
Have we considered alternative parameter names that more explicitly describe the action being taken, such as 'skipTLSVerification'? The term 'insecure' can be somewhat vague and might inadvertently give a negative impression.

Reply:
today docker uses "--insecure" flag and ORAS uses "--plain-http" flag to be set if user wants to use insecure registry. 
I and @rynowak discussed on this and decided on using `--insecure-http` to be explicit to the users that they would be using `http` link i.e `insecure` but  many of us were leaning towards using `--plain-http` flag to be consistent with ORAS and since we would be setting plainHttp flag in the backend. 
@ryanwaite wanted to know your thoughts on this.
cc: @kachawla, @AaronCrawfis , @willtsai

Reply:
I Had an offline discussion with @ryanwaite and agreed upon using `--plain-http` flag.  Will update it in the doc